### PR TITLE
[TPU] Do not cancel all jobs when one fails

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title. There's no way to cancel a single job in the workflow

cc @carmocca @borda @JackCaoG @steventk-g @Liyang90

### CODE REVIEW

File: .github/workflows/tpu-tests.yml [hunk 1]
```diff
@@ -111,21 +111,15 @@ jobs:
...
         JOB_NAME: ${{ env.PR_NUMBER }}-${{ matrix.pkg-name }}-${{ matrix.runtime }}-${{ env.SHA }}
       run: |
-        gcloud compute tpus tpu-vm create "$JOB_NAME" --accelerator-type=v4-8 --version="tpu-vm-v4-pt-$XLA_VER"
-
-    - name: Cancel job
-      if: steps.tpu-create.outcome != 'success'
-      env:
-        GH_TOKEN: ${{ github.token }}
-      run: |
-        gh run cancel ${{ github.run_id }}
```
Comment (carmocca) [inline .github/workflows/tpu-tests.yml:125]: We cannot use this because it will cancel all jobs in the workflow. Not just the one that failed

Comment (Borda) [reply .github/workflows/tpu-tests.yml:125]: Then you can open each, it is just 4 option, so please keep it

**Review by Borda — DISMISSED** [review]
Let's keep any form of canceling run to clearly separate when resources were not allowed, the actual case will silently make CI green even no tests were running because of lack of resources

**Comment by carmocca** [comment]
@Borda There's two kinds of failures when creating the TPU VM.
1. The creation couldn't work at all (maybe there's no quota, maybe a credential is invalid...). This would happen for all jobs in the workflow
2. The creation might work but capacity is limited at the moment. This could happen for 1 job in the workflow or for all of them. It's hard to say

Since cancelling a workflow cancels all jobs in the workflow, we could only use it in the first case, but not in the second one because it would force the requirement that all jobs are allocated.

Since the TPU tests are not marked as required, the actual status of the job doesn't matter.
The only benefit to adding the cancelling logic was to avoid the red X on the master CI, which the changes in this PR also avoid.

AFAIK there's no way to cancel just one job either programmatically or through the UI, so your suggestion is impossible

**Comment by Borda** [comment]
> The only benefit to adding the cancelling logic was to avoid the red X on the master CI, which the changes in this PR also avoid.

yes, but also mark all as green, which could be false positive (my suggestions are false neutral), so then pls do not expect the user's PR with affecting functionally, which will not be tested because of creation issues and merged because of all green will be responsible for it :innocent:

**Comment by carmocca** [comment]
>  so then pls do not expect the user's PR with affecting functionally, which will not be tested because of creation issues and merged because of all green will be responsible for it innocent

This is not a problem because as I said, the TPU tests are not required to pass.

If you really don't want it to show as green, then let's let it fail when the VM creation fails, which clearly signals that the tests did not run. We can have them not run in master if you care about master status.

>  i do not see any issue with canceling whole batch, it is so rare that only one job would succeed and the rest fail and even if so you still need to pass all of just one configuration

This is very wasteful. Would you cancel all cpu tests if for example one MacOS job failed to start?

**Review by awaelchli — APPROVED** [review]
I would go ahead with these changes, they are low risk and it doesn't cross over with other work or CI jobs. We just have to test a few things and see what works well for us.

**Comment by carmocca** [comment]
@Borda Can you please check this again? This is blocking my tasks

**Comment by Borda** [comment]
> I would go ahead with these changes, they are low risk and it doesn't cross over with other work or CI jobs. We just have to test a few things and see what works well for us.

Well, no other resources we use have such capacity limitations... 

> @Borda Can you please check this again? This is blocking my tasks

Sure, but I don't see any change since my last comment so what do you expect from my? Also how exactly does this cancalation blocks your work?

**Comment by carmocca** [comment]
Accidentally breaking tpu tests is fine because they are not marked as required anyways. Even if they showed as failing you could merge the PR.

I need to continue making changes to the TPU setup (https://github.com/Lightning-AI/lightning/pull/18136) so that we can more reliably test new features https://github.com/Lightning-AI/lightning/pull/18126. Since VM creation regularly fails on at least one of the jobs, the current behaviour of canceling all jobs is very wasteful and limiting

**Comment by carmocca** [comment]
I'm fine with showing them as being "failing" or "passing", you can decide that if you want. But they are not required to pass anyways so it's just a cosmetic decision

**Comment by Borda** [comment]
> Since VM creation regularly fails on at least one of the jobs, the current behaviour of canceling all jobs is very wasteful and limiting

Then drop the last job since it is regular...

**Comment by carmocca** [comment]
@Borda What do you mean?

**Comment by Borda** [comment]
> @Borda What do you mean?

You said that last job is always failing so why are you trying to run and don't remove if it regularly fails

**Comment by carmocca** [comment]
I'm saying that if you launch 4 jobs from the same workflow (this happens in parallel) and any of them fail, the failing ones will cancel the running ones.

Doesnt matter first or last. It doesnt always fail

**Comment by carmocca** [comment]
The code that runs is the same and independent. It's the environment that changes.

Just like the pl-cpu tests where we parametrize over different torch versions and so on

**Pull request merged**